src/rules/no_this.coffee

```diff
@@ -0,0 +1,16 @@
+
+module.exports = class NoThis
+    rule:
+        name: 'no_this'
+        description: '''
+        This rule prohibits 'this'.
+        Use '@' instead.
+        '''
+        level: 'ignore'
+        message: 'Don\'t use \'this\''
+
+    tokens: ['THIS']
+    lintToken: (token, tokenApi) ->
+        config = tokenApi.config[@rule.name]
+        level = config.level
+        { level, 'Replace "this" with "@"' }
```

Comment:
you rule is actually functioning in that the user would see a warning if you run it. You're attempting to pass `context` here, but just missed the key:

    { level, context: 'Replace "this" with "@"' }

level is actually optional, if you don't specify it in the rule it will default to whatever the current level is. `prefer_english_operator` is one of these special cases where it sometimes needs to pass the level explicitly.

    lintToken: (token, tokenApi) ->
        { context: 'Replace "this" with "@"' }

Sorry this is pretty much all undocumented :(

There is another convenience here you can use. you can simple return `true` and it'll get converted to an object with all the right defaults. `context` gets appended to `message` so as it is here the user will see `Don't use 'this'. Replace "this" with "@"`. In the tests you get the normalized object, so instead of checking `result.context` you can verify that `result.rule` is `no_this`